R/benchmarks.R

```diff
@@ -2,25 +2,24 @@
 #'
 #' @param run_id the run_id to get benchmarks for (default: `NULL`, list all)
 #' @param batch_id the batch_id to get benchmarks for (default: `NULL`, list all)
+#' @param run_reason a string to specify the run reason (default: `NULL`, list all)
 #' @inheritParams runs
 #'
 #' @return the response
 #' @export
-benchmarks <- function(run_id = NULL, batch_id = NULL, simplifyVector = TRUE, flatten = TRUE, ...) {
+benchmarks <- function(run_id = NULL, batch_id = NULL, run_reason = NULL, simplifyVector = TRUE, flatten = TRUE, ...) {
   req <- req_url_path_append(conbench_request(), "benchmarks")
 
+  if (!is.null(run_reason)) {
+    req <- req_url_query(req, run_reason = run_reason)
+  }
+
   if (!is.null(batch_id)) {
     req <- req_url_query(req, batch_id = paste0(batch_id, collapse = ","))
-    if (!utils::packageVersion("httr2") > "0.2.1") {
-      req$url <- curl::curl_unescape(req$url)
-    }
   }
 
   if (!is.null(run_id)) {
     req <- req_url_query(req, run_id = paste0(run_id, collapse = ","))
```

Comment:
What happens if multiple of these are non-null? Can `req_url_query()` be called multiple times additively? [It looks like you can unpack a list into it with `!!!` if you need.](https://httr2.r-lib.org/reference/req_url.html) 

If these are exclusive and you can only specify one, some assertions to ensure that are a good idea.

Reply:
> If these are exclusive and you can only specify one, some assertions to ensure that are a good idea.

This is what I ended up doing.